.github/workflows/compile-kfp.yml

```diff
@@ -0,0 +1,79 @@
+name: Compile KFP pipelines
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+  push:
+    branches: [ main ]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+
+permissions:
+  contents: read
+  actions: write
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: docling-standard
+            dir: kubeflow-pipelines/docling-standard
+            out: docling_convert_pipeline_compiled.yaml
+          - name: docling-vlm
+            dir: kubeflow-pipelines/docling-vlm
+            out: docling_convert_pipeline_compiled.yaml
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install requirements for ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          # Ensure YAML checker is available (if not already required)
+          python -c "import yaml" 2>/dev/null || pip install pyyaml
+
+      - name: Compile ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          set -euxo pipefail
+          python docling_convert_pipeline.py
+
+          # Verify output exists and is non-empty
+          test -s "${{ matrix.out }}"
+
+          # Parse YAML to ensure validity
+          python - << 'PY'
```

Comment:
`'PY` should be `EOF` here. I don't like this approach of having the script we're using to ensure validity inline within the workflow. Can we have it live somewhere else in the repository